guides/testing/testing_config.md

```diff
@@ -0,0 +1,80 @@
+# Testing Config
+
+An Oban instance's config is statically defined during initialization and it
+determines the supervision tree that Oban builds. The instance config is
+encapsulated by an `Oban.Config` struct, which is then referenced by plugins,
+queues, and most public functions.
+
+Your app's Oban config is extremely important and it deserves testing!
+
+## Testing Production Config
+
+Within the test environment, apps typically disable queues and plugins for more
+control and to prevent sandbox interference. That means only a subset of the
+full Oban config is verified when your application boots for testing.
+
+The `Oban.Config.validate/1` helper is used internally when the config
```

Comment:
> Is it really beneficial to validate configs in our tests?

In my opinion, absolutely. Test configuration and production configuration may be wildly different.

> As for whether it's configured correctly, it seems not to provide any benefits to test if queue :a is paused or not.

Consider the config for plugins with complex options, like Cron or DynamicQueues. Alternatively, queues can have involved configuration that combines global limits, rate limits, partitioning, etc.

> As for validation, I personally would prefer Oban to refuse to start when the configs are invalid, so my app wouldn't start on production and I can and have to see why in the logs.

It already does this. If the configuration is invalid the supervisor won't boot at all. That is _exactly_ the situation this aims to prevent in production. Not everybody has health checks before the app boots, and multiple people have reported a misconfiguration taking their app down.